src/generate-names.cc

```diff
@@ -429,4 +430,23 @@ Result GenerateNames(Module* module, NameOpts opts) {
   return generator.VisitModule(module);
 }
 
+std::string GenerateNameWithDollar( const std::string& func_name, Index function_index) {
```

Comment:
This function seems to duplicate functionality from `NameGenerator::GenerateName` ? Also, whitespace. Also, weird to for the start with a `$` then have to remove it again in one of the cases, that could be done more elegantly.

Reply:
NameGenerator::GenerateName requires a lot of gymnastics to be used as it is and would still generate some extra information into the names ("print_and_compute" would become "print_and_compute_b" in my tests). What I could do is to extract the common functionality from ```NameGenerator::GenerateName``` and delegate it to ```GenerateNameWithDollar```. What do you think?

Reply:
I'm not sure, hope at some point @binji  will chime in.

Reply:
Hm, I'm not sure why it would be generating a different name unless that name already exists. In that case it would disambiguate with a number, not a letter. In any case, yeah, it'd be nicer to reuse the code in `GenerateName` if possible.